kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/channel/Channel.java

```diff
@@ -73,17 +72,10 @@ public Channel(final String name, final ChannelType type, final DataType valueTy
         requireNonNull(valueType, "Channel value type cannot be null");
         requireNonNull(config, "Channel configuration cannot be null");
 
-        if (!Channel.isValidChannelName(name)) {
```

Comment:
I understand the concern. I know that we have considered efficiency over clean code. The new `channel` package is also used by `driver` package as `driver` package needs to use `ChannelRecord`, which logically means `Driver` is dependent on `Channel`s which, on the other hand, is not true as `Driver` only uses the associated configuration from `ChannelRecord` and `Driver` doesn't use `Channel`s directly. From the logical point of view that an API packaging structure provides is not clearly separated as we merged both the `AssetRecord` and `DriverRecord` into `ChannelRecord`.